### DISCUSSION TITLE
ladestand des E-Auto wird nicht angezeigt

### DISCUSSION THREAD

**Discussion by Opa62** [solectrus/solectrus#3518] [Q&A]
Hallo,

Ich habe Solectrus mit der Version 0.17 mit dem Konfigurator neu erstellt, es wird aber der Ladestand des E-Auto nicht angezeigt ???
muss man hier dann noch nachträglich in der Yaml und .env was eintragen ??
Wenn ja was und wie ??
habe meine Wollbox über MQTT evcc den Topik evcc/loadpoints/1/chargePower eingetragen.

Schon mal Vielen Dank für eure Hilfe

**Comment by ledermann** [comment]
Zwei Dinge sind zu tun:

1) Abholen des Wertes per MQTT-Collector, z.B. über so ein Mapping:

   ```
   MAPPING_X_TOPIC=evcc/loadpoints/1/chargePower
   MAPPING_X_MEASUREMENT=car
   MAPPING_X_FIELD=battery_soc
   MAPPING_X_TYPE=float
   ```

2) Sensor ergänzen: 

    ```
    INFLUX_SENSOR_CAR_BATTERY_SOC=Tesla:battery_soc
    ````

    Dies in `.env` eintragen und dann `services.dashboard.environment` ergänzen.

An einer besseren Dokumentation arbeite ich noch.

**Comment by Opa62** [comment]
services.dashboard.environment ergänzen verstehe ich jetzt nicht ganz, auch in .env eintragen , oder wie ?

**Reply by StevieC121176** [reply]
Den SOC des Fahrzeugs würde ich nicht über EVCC holen, da dieser nur aktualisiert wird wenn das Fahrzeug an der Wallbox hängt. Lieber per MQTT bereit stellen über eine Fahrzeug API oder ähnliches. Ich mache das z.Bsp. über den iobroker und MQTT

**Reply by StevieC121176** [reply]
mit dem von dir oben genannten EVCC Topic bekommst du auch nur die Ladeleistung und nicht den SOC des Autos.

**Reply by ledermann** [reply]
Ja, da stimme ich @StevieC121176 zu, das ist mir gestern nicht aufgefallen, in `evcc/loadpoints/1/chargePower` steht nicht der SOC, sondern die Ladeleistung. 

@Opa62: Du musst also woanders den Ladestand des E-Autos beschaffen. Mit ioBroker und einem für das Auto passenden Adapter könnte das gehen, das hängt aber sehr vom Auto ab.

Mit "services.dashboard.environment ergänzen" meinte ich, dass jede neue Umgebugsvariable in der `.env` auch in der `compose.yaml` aufgeführt werden muss, um einem Service den Zugriff darauf zu ermöglichen. Das ist so ein Docker-Ding: Jeder Service kann nur auf die Variablen zugreifen, die in der `compose.yaml` im Abschnitt `environment` des Service aufgelistet werden.

Ich habe gerade den Konfigurator weiter ausgebaut, sodass `E-Auto` gewählt werden kann und dann das MQTT-Topic für den Ladestand abgefragt wird. Vermutlich ist es das einfachste, du versuchst es mit dem Konfigurator erneut.

**Reply by Opa62** [reply]
Hallo zusammen, 
erstmal Vielen Dank für eure Infos, 
dann werde ich es mal versuchen über den iobroker und MQTT den Topic hin zu bringen mal schauen ob ich das schaffe,
Gibt es vieleicht ein Video auf Deutsch das erklärt wie in iobroker und MQTT einen Topic erzeugt ? 
Den iobroker habe ich schon auf den Raspberrypi installiert, aber dann wird es schwierig für mich.

Grüße
Opa62

**Reply by Opa62** [reply]
Hallo,
habe mit dem Konfigurator Selectrus neu erstellt, das Symbol Auto laden wird jetzt angezeigt, wie schon von euch erwähnt,
wir nur ein Wert angezeigt wenn ich die Wallbox anschließe.
Habe in iobroker für meinen BMW IX1 nichts gefunden, aber im Home Assistant gibt es den Adapter BMW und ich bekomme viele 
Werte angezeigt z.B. ix1_edrive20_verbleibende_batterie_in_prozent, kann ich diesen Sensor irgend wie verwenden ?
![IMG_3241](https://github.com/user-attachments/assets/c306ee1d-bcdd-44c3-8266-d4fd24f8644d)
![IMG_3242](https://github.com/user-attachments/assets/36097ee1-8214-4b6d-ab1d-ecd0698ad132)

Vielen Dank

**Reply by ledermann** [reply]
Wenn du diesen Wert zum MQTT-Broker bringen kannst, dann sollte das gehen. Für den Home Assistant gibt es das hier, um per MQTT etwas verschicken zu können:
https://www.home-assistant.io/integrations/mqtt
Möglicherweise hilft das. 

Solche Dinge sind leider ziemlich knifflig. SOLECTRUS fängt da an, wo ein Messwert per MQTT bereitsteht. Dorthin zu kommen, kann aber schwierig sein.

**Reply by Opa62** [reply]
Hallo Georg,

Vielen Dank

ich werde es versuchen
